src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -1098,7 +1083,7 @@ public override ImmutableArray<MethodSymbol> ExplicitInterfaceImplementations
                         if (!method.ContainingType.IsInterface)
                         {
                             anyToRemove = true;
-                            sawObjectFinalize = sawObjectFinalize ||
+                            sawObjectFinalize = 
```

Comment:
If you take a look at the containing loop, it's clear that `sawObjectFinalize` is always `false` in the left operand of the `||` operator. The loop exists once `sawObjectFinalize` is set to `true`.